### PR TITLE
chore: init core package

### PR DESCRIPTION
**Pull request by MrHeer**
Init `core` package.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "@x-filter/core",
+  "main": "src/index.ts",
+  "types": "src/index.ts",
+  "dependencies": {
+    "typescript": "latest"
```
Comment (thomaszdxsn) [inline packages/core/package.json:6]: why not have `pnpm.lock` & `.gitignore` files

**Review by thomaszdxsn — CHANGES_REQUESTED** [review]
lack of some files

File: packages/core/package.json (hunk 1 above)
Comment (MrHeer) [reply packages/core/package.json:6]: `.gitignore` is in the root directory.

Comment (thomaszdxsn) [reply packages/core/package.json:6]: you need install it, avoid potential conflict

Comment (MrHeer) [reply packages/core/package.json]: updated

Comment (thomaszdxsn) [reply packages/core/package.json]: and .gitignore file

Comment (MrHeer) [reply packages/core/package.json]: Not required, cause it in the root directory.

**Review by thomaszdxsn — APPROVED** [review]

**Pull request merged**